### PR TITLE
Add gitlab.com forking support

### PR DESCRIPTION
**Pull request by stepnem**
(Depends on https://github.com/magit/ghub/pull/109)

### CODE REVIEW

File: lisp/forge-gitlab.el [hunk 1]
```diff
@@ -542,6 +542,16 @@ it is all or nothing.")
   (--filter (string-match-p "\\`\\.gitlab/merge_request_templates/.+\\.md\\'" it)
             (magit-revision-files (oref repo default-branch))))
 
+(cl-defmethod forge--fork-repository ((repo forge-gitlab-repository) fork)
+  (with-slots (owner name) repo
+    (let ((resource (format "/projects/%s%%2F%s" fork name)))
+      (if (glab-get resource)
+          (forge--msg nil t nil "Repo %s/%s already exists!" fork name)
+        (forge--glab-post repo (format "/projects/%s%%2F%s/fork" owner name)
```
Comment (tarsius) [inline lisp/forge-gitlab.el]: The github variant intentionally does not check whether the fork already exists. That way `forge-fork` can be (ab)used to add a remote for a certain fork regardless of whether that fork already exists (and the user having to remember whether that is so or not).

**Comment by stepnem** [comment]
>> +      (if (glab-get resource)
> +          (forge--msg nil t nil "Repo %s/%s already exists!" fork name)
> +        (forge--glab-post repo (format "/projects/%s%%2F%s/fork" owner name)
>
> The github variant intentionally does not check whether the fork
> already exists. That way `forge-fork` can be (ab)used to add a remote
> for a certain fork regardless of whether that fork already exists (and
> the user having to remember whether that is so or not).

Yes, but precisely in order to _preserve_ that forge-fork semantics, we
do have to check this for GitLab, because she returns 4XX error (409
Conflict IIRC) if you try to create an already existing project. Letting
the error through would prevent the subsequent remote manipulation step.

**Comment by tarsius** [comment]
I overlooked that this is just a warning, not an error.

I think it would be better to avoid the additional request and just use the `:noerror` argument instead.

**Comment by stepnem** [comment]
> I overlooked that this is just a warning, not an error.
>
> I think it would be better to avoid the additional request and just use the `:noerror` argument instead.

Indeed, I didn't realize I could use :noerror, thanks.

**Comment by tarsius** [comment]
Thanks!

**Pull request merged**